### PR TITLE
Added CLI based integration test

### PR DESCRIPTION
**Pull request by ruks**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -293,6 +293,11 @@
                 <artifactId>lib-creator</artifactId>
                 <version>${ballerina.platform.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.wso2.am.microgw</groupId>
+                <artifactId>org.wso2.micro.gateway.cli</artifactId>
+                <version>${wso2.apim.cli.version}</version>
+            </dependency>
         </dependencies>
```
Comment (Rajith90) [inline pom.xml]: We already have thiss on root pom

File: pom.xml [hunk 2]
```diff
@@ -348,6 +353,7 @@
 
     <properties>
         <ballerina.platform.version>0.974.0</ballerina.platform.version>
+        <wso2.apim.cli.version>2.5.0-SNAPSHOT</wso2.apim.cli.version>
```
Comment (Rajith90) [inline pom.xml]: Do we need another version here

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/throttle_filter.bal [hunk 3]
```diff
@@ -63,8 +63,11 @@ public type ThrottleFilter object {
             AuthenticationContext keyvalidationResult = check <AuthenticationContext>context.attributes[
             AUTHENTICATION_CONTEXT];
             requestFilterResult = {canProceed:true};
-            if (isSubscriptionLevelThrottled(context, keyvalidationResult)){
-                if (keyvalidationResult.stopOnQuotaReach) {
+            boolean isThrottled;
+            boolean stopOnQuata;
+            (isThrottled, stopOnQuata) = isSubscriptionLevelThrottled(context, keyvalidationResult);
+            if (isThrottled){
```
Comment (tharikaGitHub) [inline components/micro-gateway-core/src/main/ballerina/gateway/filters/throttle_filter.bal]: Shall we fix the formatting here?

File: tests/src/test/java/org/wso2/micro/gateway/tests/common/MockAPIPublisher.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+        JSONObject apiJsonObj = new JSONObject(apiJson);
+        apiJsonObj.put("endpointConfig", endpoint.toString());
+        apiJsonObj.put("apiDefinition", apiDefinition);
+        apiJsonObj.put("name", api.getName());
+        apiJsonObj.put("version", api.getVersion());
+        apiJsonObj.put("context", api.getContext());
+        apiJsonObj.put("provider", api.getProvider());
+
+        //        private String[] tiers;
```
Comment (tharikaGitHub) [inline tests/src/test/java/org/wso2/micro/gateway/tests/common/MockAPIPublisher.java]: Shall we remove commented out lines here?

File: tests/src/test/java/org/wso2/micro/gateway/tests/services/APIInvokeTestCase.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
+package org.wso2.micro.gateway.tests.services;
```
Comment (Rajith90) [inline tests/src/test/java/org/wso2/micro/gateway/tests/services/APIInvokeTestCase.java]: package should be below the licensce

**Pull request merged**